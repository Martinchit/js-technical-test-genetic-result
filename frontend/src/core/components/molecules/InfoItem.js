import React from 'react';
import PropTypes from 'prop-types';
import { FlexBox } from '../atoms/FlexBox';
import { Text } from '../atoms/Text';
import { Spacer } from '../atoms/Spacer';
import { convertInfoItemData } from '../../lib/utils/dataConversion';

export const InfoItem = ({ header, description }) => {
  return (
    <FlexBox flexDirection="column" alignItems="flex-start">
      <Text fontSize={1.5} fontColor="grey">
        {header}
      </Text>
      <FlexBox flexDirection="column" alignItems="flex-start" margin="3px 1px">
        {Array.isArray(description) ? (
          description.map((t) => (
            <Text fontSize={1.2} key={t}>
              {convertInfoItemData(header, t)}
            </Text>
          ))
        ) : (
          <Text fontSize={1.2}>{convertInfoItemData(header, description)}</Text>
        )}
      </FlexBox>
      <Spacer />
    </FlexBox>
  );
};

InfoItem.propTypes = {
  header: PropTypes.string,
  description: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
};

InfoItem.defaultProps = {
  header: '',
  description: '',
};
