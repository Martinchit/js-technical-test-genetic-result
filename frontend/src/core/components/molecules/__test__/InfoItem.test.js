import React from "react";
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { InfoItem } from "../InfoItem";
import { Theme } from "../../../lib/theme/globalStyle";

describe("InfoItem", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<InfoItem />)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders correctly with array description", () => {
    const description = ['testOne', 'testTwo']
    const wrapper = shallow(<InfoItem description={description} />)
    expect(wrapper).toMatchSnapshot()
  })
})