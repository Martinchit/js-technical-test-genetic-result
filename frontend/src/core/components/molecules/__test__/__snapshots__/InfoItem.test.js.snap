// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoItem renders correctly 1`] = `ShallowWrapper {}`;

exports[`InfoItem renders correctly with array description 1`] = `ShallowWrapper {}`;
